src/client/containers/authentication/authentication.js

```diff
@@ -22,33 +24,65 @@ class Authentication extends Component {
 
     onLinkClick = () => {
       const { isMember } = this.state;
-      this.setState({ isMember: !isMember });
+      this.setState({ isMember: !isMember, userData: {}, validationErrors: {} });
     };
 
-    onFormSubmit = (submit) => {
+    onFormSubmit = async (submit) => {
       submit.preventDefault();
-      const { userData: { username, email, password }, isMember } = this.state;
+      const { userData, isMember } = this.state;
+      const {
+        username, email, password, passwordConfirmation,
+      } = userData;
       const { history, onAuth } = this.props;
       let path;
       let data;
 
       if (!isMember) {
         path = '/registration';
-        data = { username, email, password };
+        data = {
+          username,
+          email,
+          password,
+          passwordConfirmation,
+        };
       } else {
         path = '/login';
-        data = { email, password };
+        data = {
+          email,
+          password,
+        };
+      }
+      const validationRes = formValidation(data);
+      if (Object.keys(validationRes).length === 0) {
+        const serverError = await onAuth(path, data, history);
+        this.setState({ validationErrors: serverError });
+      } else {
+        this.setState({ validationErrors: validationRes });
       }
-      onAuth(path, data, history);
     };
 
-    filterFields = (arr, status) => (status ? arr.filter(el => status === el.isMember) : arr);
+    contentToRender = (fieldsinfo, statusMember) => {
+      let content;
+      if (statusMember) {
+        content = {
+          fildsToRender: fieldsinfo.filter(el => statusMember === el.isMember),
+          formTitle: 'Login',
+          message: 'Dont have an account? Register!',
+        };
+      } else {
+        content = {
+          fildsToRender: fieldsinfo,
+          formTitle: 'Login',
+          message: 'Dont have an account? Register!',
+        };
+      }
+
+      return content;
```

Comment:
repetition, return object & use variable with ternary only for "fildsToRender"

Reply:
done